### ISSUE TITLE
Load_As_Byte_Array does not load modules

### ISSUE THREAD

**Issue by Trojaner** [SmartlyDressedGames/Unturned-3.x-Community#2338]
![image](https://user-images.githubusercontent.com/1809172/105122002-72cdcd00-5ae6-11eb-8274-7c3e3fe072ae.png)

It looks like typeof(IModuleNexus).IsAssignableFromType(moduleType) will return false when Load_As_Byte_Array is enabled. As a result of this modules using this will not load at all. 

The only logical explanation for this issue is that it binds to another instance of Assembly-CSharp.dll that was loaded using Assembly.Load when OnAssemblyResolve is called.

There is no way to workaround this.

**Comment by SDGNelson** [comment]
Did you start using this when it was added back in September? If you have a proposed solution I am open to considering it, but from the gap between request and using I am inclined towards removing it and suggesting you go with the approach in your OpenMod issue.

**Comment by Trojaner** [comment]
No, we just started using this with OpenMod 3 as the `/openmod upgrade` command was just added.

**Comment by Trojaner** [comment]
I did some investigation and was not able to figure out why `typeFromHandle.IsAssignableFrom(type)` returns false all the time. Interestingly, `typeof(IModuleHook) == type.GetInterfaces().First()` returns true. At this point I will just assume that this is actually a mono bug.

I added the mentioned workaround to OpenMod so the issue is fixed for us already. If you still want to support Load_As_Byte_Array, a non-breaking workaround would be replacing the `typeFromHandle.IsAssignableFrom(type)` call with:
```cs
type.GetInterfaces().Any(i => i == typeFromHandle); 
```
